packages/flutter_tools/lib/src/localizations/gen_l10n_types.dart

```diff
@@ -331,13 +335,27 @@ class Message {
       );
     }
 
-    final RegExpMatch? pluralRegExp = _pluralRE.firstMatch(_value(bundle, resourceId));
-    final bool isPlural = pluralRegExp != null && pluralRegExp.groupCount == 1;
-    if (attributes == null && isPlural) {
-      throw L10nException(
-        'Resource attribute "@$resourceId" was not found. Please '
-        'ensure that plural resources have a corresponding @resource.'
-      );
+    if (attributes == null) {
+      final RegExpMatch? pluralRegExp = _pluralRE.firstMatch(
```

Comment:
There seems to be a lot of duplicated logic for `plural` & `select` validation. Worth to make it a helper method that takes the regex?

Reply:
Yes, done.